swabbie-aws/src/test/java/com/netflix/spinnaker/swabbie/aws/autoscalinggroups/ZeroInstanceDisabledServerGroupRuleTest.kt

```diff
@@ -16,11 +16,17 @@
 
 package com.netflix.spinnaker.swabbie.aws.autoscalinggroups
 
+import com.amazonaws.services.autoscaling.model.SuspendedProcess
 import org.junit.jupiter.api.Assertions
 import org.junit.jupiter.api.Test
+import java.time.Clock
 import java.time.Instant
+import java.time.temporal.ChronoUnit
 
 object ZeroInstanceDisabledServerGroupRuleTest {
+
+  private val clock = Clock.systemUTC()
```

Comment:
Clocks in tests should use our [`MutableClock`](https://github.com/spinnaker/kork/blob/master/kork-test/src/main/java/com/netflix/spinnaker/kork/test/time/MutableClock.java) so that we can set and increment them. We don't want to be relying on a system clock.

Reply:
That goes for every test clock in this PR.
If you don't need to increment the clock at all, you can use the `Clock.fixed(...)` method.

Reply:
I think a system clock is fine if you don’t need to mutate the clock to test a piece of functionality.
But this requires your test to use this clock everywhere to make relative calculations of time, because the class under test relies on that same clock.

However looking more in the test, this clock isn’t used everywhere: this makes test brittle if you are using LocalDateTime.now() and not use the clock to match the class under test.

Reply:
I think any use of the actual clock is a recipe for tests to become brittle, fail at certain times, hard to diagnose. I'd much rather use a fixed clock. It's just as easy and much safer.